lib/puppet/type/service.rb

```diff
@@ -253,6 +255,49 @@ def sync
       end
     end
 
+    newproperty(:logonaccount, :required_features => :configurable_logon_credentials) do
+      desc "Specify an account for service logon on Windows"
+
+      munge do |value|
+        return value if value == 'LocalSystem'
+
+        # @account="Luchi",
+        # @account_type=:SidTypeUser,
+        # @domain="DESKTOP-P1OLRES",
+        # @domain_account="DESKTOP-P1OLRES\\Luchi",
+        # @sid="S-1-5-21-1825778692-957321060-2432639842-1000",
+        # @sid_bytes=[1, 5, 0, 0, 0, 0, 0, 5, 21, 0, 0, 0, 4, 44, 211, 108, 100, 143, 15, 57, 98, 35, 255, 144, 232, 3, 0, 0]>
+        account = Puppet::Util::Windows::SID.name_to_principal(value).domain_account
```

Comment:
Note to self: 
`Puppet::Util::Windows::SID.name_to_principal("Luchi")` will find user
`Puppet::Util::Windows::SID.name_to_principal("DESKTOP-P1OLRES\\Luchi")` will find user
but `Puppet::Util::Windows::SID.name_to_principal(".\\Luchi")` won't find user